zerver/worker/queue_processors.py

```diff
@@ -480,6 +482,17 @@ def get_bot_api_client(self, user_profile):
 
     def get_bot_handler(self, service):
         # type: (Service) -> Any
+
+        # Convert unicode service.name to string.
+        str_service_name = unicodedata.normalize('NFKD', service.name).encode('ascii', 'ignore')
+
+        # Assert if this service is present in EMBEDDED_BOT_SERVICES.
+        assert EmbeddedBotService(str_service_name) in EMBEDDED_BOT_SERVICES
+
+        for embedded_bot_service in EMBEDDED_BOT_SERVICES:
+            if embedded_bot_service.name == service.name:
+                assert embedded_bot_service.IS_ALLOWED is True
+
         bot_module_name = 'zulip_bots.bots.%s.%s' % (service.name, service.name)
         bot_module = importlib.import_module(bot_module_name)  # type: Any
         return bot_module.handler_class()
```

Comment:
This function should probably have tests; ideally, the tests would have caught the mispelling of `converter` above.  

Since this function doesn't access `self`, I think we probably want to move the code in this function into a separate function in `zerver/lib/bot_lib.py` and fully unit-test that function.  Tests could look like this:
```
converter = get_bot_handler("converter")
self.assertEqual(str(type(converter)), "ConverterBot") # I have no idea what the correct value is here, but some sort of test like this would be good

with self.assertRaises(...)
    converter = get_bot_handler("invalid")

```